types/document.d.ts

```diff
@@ -0,0 +1,244 @@
+import mongodb = require('mongodb');
+
+declare module 'mongoose' {
+
+  /** A list of paths to skip. If set, Mongoose will validate every modified path that is not in this list. */
+  type pathsToSkip = string[] | string;
+
+  class Document<T = any, TQueryHelpers = any, DocType = any> {
+    constructor(doc?: any);
```

Comment:
Can this be

```suggestion
    constructor(doc?: Partial<T>);
```

?

Reply:
Unfortunately not. Probably more correct to be `doc ?: AnyObject`